airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/GlueOperations.java

```diff
@@ -0,0 +1,156 @@
+                        .withTableType("GOVERNED")
+                        .withStorageDescriptor(
+                            new StorageDescriptor()
+                                .withLocation(location)
+                                .withColumns(transformSchema(jsonSchema))
+                                .withInputFormat("org.apache.hadoop.mapred.TextInputFormat")
+                                .withOutputFormat("org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat")
+                                .withSerdeInfo(
+                                    new SerDeInfo()
+                                        .withSerializationLibrary("org.openx.data.jsonserde.JsonSerDe")
+                                        .withParameters(Map.of("paths", ","))
+                                )
+                        )
+                        .withPartitionKeys(List.of())
+                        .withParameters(Map.of("classification", "json"))
+                );
+
+            awsGlueClient.createTable(createTableRequest);
+        }
+    }
+
+    @Override
+    public void deleteTable(String databaseName, String tableName) {
+
+        DeleteTableRequest deleteTableRequest = new DeleteTableRequest()
+            .withDatabaseName(databaseName)
+            .withName(tableName);
+
+        awsGlueClient.deleteTable(deleteTableRequest);
+
+    }
+
+    public Collection<Column> transformSchema(JsonNode jsonSchema) {
+
+        UnaryOperator<String> typeMapper = type -> switch (type) {
+            case "string" -> "string";
+            case "number" -> "float";
+            // TODO (itaseski) does the airbyte schema support integers as first class types or as airbyte_types?
+            case "integer" -> "int";
+            case "boolean" -> "boolean";
+            // TODO (itaseski) throw exception on unknown types or set as received?
+            default -> type;
+        };
+
+        Map<String, JsonNode> properties = objectMapper.convertValue(jsonSchema.get("properties"), new TypeReference<>() {});
+
+        return properties.entrySet().stream()
+            .map(es -> {
+                String type = filterTypes(es.getValue().get("type")).iterator().next();
+                return switch (type) {
+                    case "array" -> {
+                        Set<String> types = filterTypes(es.getValue().get("items").get("type"));
+                        if (types.size() > 1) {
+                            yield new Column().withName(es.getKey()).withType("array<string>");
+                        } else {
+                            yield new Column().withName(es.getKey())
+                                .withType("array<" + typeMapper.apply(types.iterator().next()) + ">");
+                        }
+                    }
+                    case "object" -> {
+                        //TODO (itaseski) should we take into account nested objects and generate nested structs i.e struct<col_name : struct<col_name : string>>
```

Comment:
Unfortunately structs only support primitive data types. This means it will likely require some means of flattening nested records. Possibly by allowing a configurable separator (e.g. `__`).

However, now that I type this, that means that the actual data would need to be mutated to represent the flattened schema structure. Alternatively, anything that's nested deeper than 1 level would need to be treated as a string and parsed as JSON in the query engine.

Reply:
After all it seems that Glue does support nesting of complex types. I have fully reimplemented the schema generation algorithm to use recursion in order to include all the nested types in the schema

Reply:
After all it seems that Glue does support nesting of complex types. I have fully reimplemented the schema generation algorithm to use recursion in order to include all the nested types in the schema